lib/improver/nbhood/circular_kernel.py

```diff
@@ -0,0 +1,361 @@
+        """
+        Method to apply a circular kernel to the data within the input cube in
+        order to smooth the resulting field.
+
+        Parameters
+        ----------
+        fullranges : Numpy.array
+            Number of grid cells in all dimensions used to create the kernel.
+            This should have the value 0 for any dimension other than x and y.
+        ranges : Tuple
+            Number of grid cells in the x and y direction used to create
+            the kernel.
+        weighted_mode : Boolean
+            True when a weighted circular kernel is required.
+            False will return a kernel consisting only of ones and zeroes.
+
+        Returns
+        -------
+        kernel : Numpy.array
+            Array containing the circular smoothing kernel.
+            This will have the same number of dimensions as fullranges.
+
+        """
+        # Define the size of the kernel based on the number of grid cells
+        # contained within the desired radius.
+        kernel = np.ones([int(1 + x * 2) for x in fullranges])
+        # Create an open multi-dimensional meshgrid.
+        open_grid = np.array(np.ogrid[tuple([slice(-x, x+1) for x in ranges])])
+        if weighted_mode:
+            # Create a kernel, such that the central grid point has the
+            # highest weighting, with the weighting decreasing with distance
+            # away from the central grid point.
+            open_grid_summed_squared = np.sum(open_grid**2.).astype(float)
+            kernel[:] = (
+                (np.prod(ranges) - open_grid_summed_squared) / np.prod(ranges))
+            mask = kernel < 0.
+        else:
+            mask = np.reshape(
+                np.sum(open_grid**2) > np.prod(ranges), np.shape(kernel))
+        kernel[mask] = 0.
+        return kernel
+
+
+class CircularProbabilities(object):
+
+    """
+    Methods for use in the calculation and application of a circular
+    neighbourhood.
+
+    A maximum kernel radius of 500 grid cells is imposed in order to
+    avoid computational ineffiency and possible memory errors.
+    """
+
+    def __init__(self, weighted_mode=True, percentiles=None):
+        """
+        Initialise class.
+
+        Parameters
+        ----------
+        weighted_mode : boolean
+            If False, use a circle with constant weighting.
+            If True, use a circle for neighbourhood kernel with
+            weighting decreasing with radius.
+
+        percentiles : list (optional)
+            This is included to allow a standard interface for both the
+            percentile and probability neighbourhood plugins.
+        """
+        self.weighted_mode = weighted_mode
+        self.percentiles = percentiles
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        result = ('<CircularProbabilities: weighted_mode: {}>')
+        return result.format(self.weighted_mode)
```

Comment:
The percentiles argument is not used by the module and is only included to make the plugin simpler. It therefore seems irrelevant to write out the value.